docs/getting-started-tutorial/4-css-selector-transforms.adoc

```diff
@@ -0,0 +1,171 @@
+:idprefix:
+:idseparator: -
+:toc: right
+:toclevels: 2
+
+# CSS Selector Transforms
+
+Because Lift operates by transforming HTML trees, it needs an easy way to
+specify those transformations. Otherwise we'd be doing a bunch of recursive
+tree searches and munges and it would get ugly and probably slow to boot. To
+deal with transformations easily, we use a small subset of CSS selectors, with
+a few Lift idiosyncrasies to maximize performance and address some use cases
+that are particularly useful when transforming trees.
+
+We'll leave forms for the next section, as forms always come with a catalog of
+related functionality, and focus on binding the list of chat messages in this
+section. We'll also add a new one before every page load, so that we can see
+the list changing.
+
+First, we'll define a variable to hold the messages:
+
+```
+...
+object Chat {
+  var messageEntries = List[String]()
+...
+}
+```
+
+Then, we can change the definition of the `messages` method to bind the
+contents of the message list:
+
+```
+...
+
+import net.liftweb.util.Helpers._
+
+...
+  def messages = {
+    "li *" #> messageEntries
+  }
+...
+```
+
+In the previous section, we mentioned that Lift snippets can return
+`(NodeSeq)=>NodeSeq` functions. That is what's happening here: Lift's CSS
+selector transforms are actually functions that take a `NodeSeq` and return a
+`NodeSeq`, constructed using an easy-to-read syntax.
+
+What we do in this particular transformation is select all ``li``s. We then
+specify that we want to transform them by replacing their contents (`*`) by
+whatever is on the right. The right side, however, is a list, in this case of
+``String``s. When there's a list on the right side of a transformation, Lift
+repeats the matched element or elements once for each entry in the list, and
+binds the contents of each element in turn.
+
+Let's start up Lift and see what's going on. In your terminal, enter the
+directory of the chat app and start up the application:
+
+```
+$ sbt
+> container:start
```

Comment:
We'll need to update this to `jetty:start` once we upgrade to the new version of the web plugin